src/WebJobs.Script.WebHost/Health/TokenExpirationService.cs

```diff
@@ -13,54 +14,92 @@
 
 namespace Microsoft.Azure.WebJobs.Script.WebHost.Health
 {
-    internal class TokenExpirationService : BackgroundService
+    internal class TokenExpirationService : IHostedService, IDisposable
```

Comment:
Why the switch away from `BackgroundService`?

Reply:
I think Fabio mentioned on the call to make this like the assembly analysis service: https://github.com/Azure/azure-functions-host/blob/dev/src/WebJobs.Script.WebHost/AssemblyAnalysis/AssemblyAnalysisService.cs

Reply:
Either way here is fine.

Reply:
Yeah I formatted it like AssemblyAnalysisService initially, but since the `BackgroundService` method has been tested pretty throughly already I'll stick to that with adding the try/catch to the implementation of `ExecuteAsync`